File: h2database/src/main/java/p/k/tools/h2db/LogRowMapper.java
package p.k.tools.h2db;

import org.springframework.jdbc.core.RowMapper;

import java.sql.ResultSet;
import java.sql.SQLException;

public class LogRowMapper implements RowMapper {

    public Object mapRow(ResultSet rs, int index) throws SQLException {
        LogRecord logRecord = new LogRecord(rs.getString("username"), rs.getString("password"), rs.getInt("age"));
        logRecord.setId(rs.getLong("id"));
        return logRecord;

    }
}

File: h2database/src/main/java/p/k/tools/h2db/LogRowMapper.kt
package p.k.tools.h2db

import org.springframework.jdbc.core.RowMapper

import java.sql.ResultSet
import java.sql.SQLException

class LogRowMapper : RowMapper<*>
{

    @Throws(SQLException::class)
    override fun mapRow(rs: ResultSet, index: Int): Any
    {
        val logRecord = LogRecord(rs.getString("username"), rs.getString("password"), rs.getInt("age"))
        logRecord.id = rs.getLong("id")
        return logRecord

    }
}